tests/unit/s2n_client_cert_verify_test.c

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <stdint.h>
+#include <s2n.h>
+
+#include "s2n_test.h"
+#include "testlib/s2n_testlib.h"
+
+#include "tls/s2n_tls.h"
+#include "tls/s2n_connection.h"
+#include "utils/s2n_safety.h"
+
+const uint8_t test_signature_data[] = "I signed this";
+const uint32_t test_signature_size = sizeof(test_signature_data);
+const uint32_t test_max_signature_size = 2 * sizeof(test_signature_data);
+
+static int test_size(const struct s2n_pkey *pkey)
+{
+    return test_max_signature_size;
+}
+
+static int test_sign(const struct s2n_pkey *priv_key, s2n_signature_algorithm sig_alg,
+        struct s2n_hash_state *digest, struct s2n_blob *signature)
+{
+    memcpy_check(signature->data, test_signature_data, test_signature_size);
```

Comment:
Nit: We could use `EXPECT_MEMCPY_SUCESS` instead here as it is the test file.

Reply:
The two behave differently. EXPECT_MEMCPY_SUCCESS causes the test to immediately exit here, on line 38. memcpy_check instead follows normal error patterns and could theoretically be handled / ignored / modified by the caller. Be careful about using EXPECT_ methods in test implementations of methods / callbacks.